CleverPushLocation/Source/CleverPushLocation.m

```diff
@@ -82,17 +107,53 @@ + (void)locationManager:(CLLocationManager *)manager didDetermineState:(CLRegion
     [CPLog info:@"LocationManager: LocationManager didDetermineState %@", [region identifier]];
     if (state == CLRegionStateInside) {
         [self locationManager:manager didEnterRegion:region];
+    }  else if (state == CLRegionStateOutside) {
+        [self locationManager:manager didExitRegion:region];
     }
 }
 
 + (void)locationManager:(CLLocationManager *)manager didEnterRegion:(CLRegion *)region {
     [CPLog info:@"LocationManager: Entered Geo Fence %@", [region identifier]];
-    [self trackGeoFence:[region identifier] withState:@"enter"];
+    if (geoFenceTimeoutCompleted  == false) {
+        [geoFenceTimer invalidate];
+        geoFenceTimerDelay = 0;
+        geoFenceTimer = [NSTimer scheduledTimerWithTimeInterval:geoFenceTimerInterval
+                                                         target:self
+                                                       selector:@selector(geoFenceHandleTimer:)
+                                                       userInfo:nil
+                                                        repeats:YES];
+    } else {
+        [self trackGeoFence:[region identifier] withState:geoFenceEnterState];
+    }
 }
 
 + (void)locationManager:(CLLocationManager *)manager didExitRegion:(CLRegion *)region {
     [CPLog info:@"LocationManager: Exited Geo Fence %@", [region identifier]];
-    [self trackGeoFence:[region identifier] withState:@"exit"];
+    if (geoFenceTimeoutCompleted == false) {
+        [geoFenceTimer invalidate];
+        geoFenceTimerDelay = 0;
+        geoFenceTimer = [NSTimer scheduledTimerWithTimeInterval:geoFenceTimerInterval
+                                                         target:self
+                                                       selector:@selector(geoFenceHandleTimer:)
+                                                       userInfo:nil
+                                                        repeats:YES];
+    } else {
+        [self trackGeoFence:[region identifier] withState:geoFenceExitState];
+    }
+}
+
++ (void)geoFenceHandleTimer: (NSTimer *) Timer {
```

Comment:
Code Style:
no space after `: ` and no space after `(NSTimer *)`
variable rename Timer -> timer